src/java.base/share/classes/java/util/Map.java

```diff
@@ -817,10 +817,9 @@ default void replaceAll(BiFunction<? super K, ? super V, ? extends V> function)
      *
      * @param key key with which the specified value is to be associated
      * @param value value to be associated with the specified key
-     * @return the previous value associated with the specified key, or
-     *         {@code null} if there was no mapping for the key.
-     *         (A {@code null} return can also indicate that the map
-     *         previously associated {@code null} with the key,
+     * @return {@code null} if the specified key was considered absent, else returns
```

Comment:
"Considered" feels out of place. No other method in Map uses it. Try to rephrase that sentence or, if it helps, the complete `@return` tag. (@stuart-marks might have more substantial feedback.)

Reply:
Yeah, I wasn't sure about that, I can make it more specific, I used `considered` here because both unmapped and key maps to `null` is considered to be absent.

Reply:
I think `absent or {@code null}` is no less concise yet it is way more accurate than `considered absent`. So something like `@return {@code null} if the mapping for the specified key is absent or has a {@code null} value`?

Reply:
I used your wording @liach but changed "has" to past tense (as `putIfAbsent` will have replaced the previous `null` value).